app/controllers/votes_controller.rb

```diff
@@ -0,0 +1,41 @@
+class VotesController < ApplicationController
+  def upvote
+    found = Votes.find_by(user: params[:user], api: params[:api_id])
+
+    if found == nil
+      Votes.create(
+        user: User.find(params[:user]),
+        api: Api.find(params[:api_id]),
+        user_vote: true)
+      render json: Votes.calculate_votes(params[:api_id])
+      return
+    end
+    if found.user_vote == false
+      found.user_vote = true
+      found.save
+      render json: Votes.calculate_votes(params[:api_id])
+      return
+    end
+    render json: Votes.calculate_votes(params[:api_id])
+  end
+
+  def downvote
+    found = Votes.find_by(user: params[:user], api: params[:api_id])
+
+    if found == nil
+      Votes.create(
+        user: User.find(params[:user]),
+        api: Api.find(params[:api_id]),
+        user_vote: false)
+      render json: Votes.calculate_votes(params[:api_id])
+      return
+    end
+    if found.user_vote == true
+      found.user_vote = false
+      found.save
+      render json: Votes.calculate_votes(params[:api_id])
+      return
+    end
+    render json: Votes.calculate_votes(params[:api_id])
+  end
+end
```

Comment:
The `VotesController` probably needs to be factored out into an `API` namespace. Since all your other controller render html views and do not return JSON, this one is out of place being among them. You want to separate the API interfaces of your application out from the rest. [Here's](https://github.com/mlg-/surrealist_poetry_api) an example. Check out how the controllers folder has an API/v1 folder (a namespace is what you would call this, similar to how your views related to particular models are namespaced to a folder with the same name). Note that using `API` instead of `Api` requires an override in the [inflections](https://github.com/mlg-/surrealist_poetry_api/blob/master/config/initializers/inflections.rb) file.